src/main/java/org/prebid/server/bidder/applogy/ApplogyBidder.java

```diff
@@ -112,7 +90,7 @@ private Imp processImp(Imp imp) {
         Banner banner = imp.getBanner();
         if (banner != null) {
             if (banner.getH() == null || banner.getW() == null || banner.getH() == 0 || banner.getW() == 0) {
-                if (banner.getFormat() == null) {
+                if (banner.getFormat() == null || CollectionUtils.isEmpty(banner.getFormat())) {
```

Comment:
Please check the `CollectionUtils.isEmpty()` method. It is a null safe check of whether the given collection is empty, meaning that you don't need to check is passed collection is not null - that check is inside, otherwise you would just call `banner.getFormat().isEmpty()` after null check (which what that utility method does).

Just leave `CollectionUtils.isEmpty(banner.getFormat())` for the `if` condition.